x-pack/apm-server/aggregation/txmetrics/aggregator.go

```diff
@@ -218,12 +227,23 @@ func (a *Aggregator) publish(ctx context.Context, period time.Duration) error {
 			delete(svcEntry.m, hash)
 		}
 		if svcEntry.other != nil {
+			overflowCount := int64(svcEntry.otherCardinalityEstimator.Estimate())
+			if isMetricsPeriod {
+				activeGroups = int64(current.entries)
+				if svc == overflowBucketName {
+					servicesOverflow += overflowCount
+				} else if svcEntry.entries >= a.config.MaxTransactionGroupsPerService {
+					perSvcTxnGroupsOverflow += overflowCount
+				} else {
+					txnGroupsOverflow += overflowCount
+				}
+			}
```

Comment:
[**For reviewers**] Collection of metrics in `publish` method was driven by the following factors:

1. Collect total number of unique transaction groups that overflowed due to _max services limit_, _per service max transaction limit_, and _max transaction limit_. To do this, we would require the estimate from the hyperloglog sketch. While, the value would still be an estimate due to hash-collisions and hyperloglog usage but the metrics will be better aligned with the published metricset.
2. Hyperloglog++ sketch with sparse representation requires a write lock even for estimating cardinality.
3. IIUC, `CollectMonitoring` is part of a pull-based monitoring architecture, and doing a full lock on the whole struct didn't sound like a good idea.